iot/iot-device-registry-infinispan/src/main/java/io/enmasse/iot/registry/infinispan/util/Credentials.java

```diff
@@ -155,7 +158,28 @@ private static JsonArray mapSecrets(final DeviceCredential credential) {
         return credentials
                 .stream()
                 .map(Credentials::fromInternal)
+                .map(Credentials::removePasswordDetails)
```

Comment:
IIRC this will be called every time we read the credentials. If that is the case, I think we need to perform this when the credentials are stored. Which happens much less frequently than the read operation.

Reply:
@ctron  we need to keep those details in the registry ! And this is called on read, but only from the management side. so that is not in the adapter cache critical read path :)